src/radon/index.js

```diff
@@ -1,761 +1,1181 @@
+    const lastOperatorOutputType = Array.isArray(lastOperator.selected.output_type)
+      ? lastOperator.selected.output_type[0]
+      : lastOperator.selected.output_type
+    const outputType = TYPES[lastOperatorOutputType.toUpperCase()]
+    const options = Object.entries(typeSystem[outputType]).map(option => {
+      return {
+        markup: 'option',
+        hierarchical_type: 'operator_option',
+        label: option[0],
+        output_type: option[1][1][0],
+      }
+    })
+    const operatorNumber = Object.values(typeSystem[outputType])[0][0]
+    const operatorInfo = OPERATOR_INFOS[operatorNumber]
+
+    const args = operatorInfo.arguments.map((argument, index) => {
+      if ([TYPES.BOOLEAN, TYPES.INTEGER, TYPES.FLOAT, TYPES.STRING].includes(argument.type)) {
+        return {
+          id: generateId(),
+          label: argument.name,
+          markup_type: 'input',
+          hierarchical_type: 'argument',
+          value: '',
+        }
+      } else if (argument.type === TYPES.MAPPER) {
+        // TODO: return correct markup
+        return {
+          id: generateId(),
+          label: argument.name,
+          markup_type: 'input',
+          hierarchical_type: 'argument',
+          value: '',
+        }
+      } else if (argument.type === TYPES.REDUCER) {
+        return {
+          id: generateId(),
+          label: argument.name,
+          markup_type: 'input',
+          hierarchical_type: 'argument',
+          selected: {
+            markup_type: 'option',
+            hierarchical_type: 'operator_option',
+            label: Object.keys(REDUCERS)[0],
+            output_type: 'array',
+          },
+          options: MARKUP_REDUCER_OPTIONS,
+        }
+      } else if (argument.type === TYPES.FILTER) {
+        return {
+          id: generateId(),
+          label: argument.name,
+          markup_type: 'input',
+          hierarchical_type: 'argument',
+          selected: {
+            markup_type: 'option',
+            hierarchical_type: 'operator_option',
+            label: Object.keys(FILTERS)[0],
+            output_type: 'array',
+          },
+          options: MARKUP_FILTER_OPTIONS,
+        }
+      } else {
+        console.log('You are using an operator with a non supported argument type')
+      }
+    })
+
+    const selected = {
+      hierarchical_type: 'operator_option',
+      label: operatorInfo.name,
+      arguments: args,
+      output_type: outputType,
+    }
+
+    const newOperator = {
+      id: generateId(),
+      markup_type: 'select',
+      hierarchical_type: 'operator',
+      output_type: outputType,
+      selected: selected,
+      options: options,
+    }
+
+    if (stage === 'retrieve') {
+      this.markup[stage][sourceIndex].script.push(newOperator)
+    } else {
+      this.markup[stage].script.push(newOperator)
+    }
+    return newOperator
+  }
+
+  updateElement(id, value, stage) {
+    const searchAndReplace = function(currentMarkup, id, value, found) {
```

Comment:
`found` is always false, and instead of setting `found` to true you can just not call the function.
```suggestion
    const searchAndReplace = function(currentMarkup, id, value) {
```

Reply:
And this entire function is very ineficient, you want a map of `id => markup`.